src/pip/_vendor/requests/compat.py

```diff
@@ -8,7 +8,10 @@
 Python 3.
 """
 
-from pip._vendor import chardet
+try:
+    from pip._vendor import chardet
+except ImportError:
+    import charset_normalizer as chardet
```

Comment:
Should we patch this to remove the try-except block? I feel it's more predictable if we always import the vendored chardet.

Reply:
I've just patched it.

I thought about an alternative: substituting all `charset_normalizer` imports with `pip._vendor.charset_normalizer` using vendoring capabilities. This way we won't have to maintain the additional patches, and switching to `charset_normalizer` will be smoother.

Reply:
That sounds like a good idea. Would it be easier if I merge this first or can you do that in this PR?

Reply:
On second thought, let’s have this in main so we can more easily evaluate this alternative solution.